hypothesis-python/src/hypothesis/internal/conjecture/providers.py

```diff
@@ -827,3 +831,38 @@ def draw_bytes(
     ) -> bytes:
         values = self._draw_collection(min_size, max_size, alphabet_size=2**8)
         return bytes(values)
+
+
+class URandom(Random):
+    # we reimplement a Random instance instead of using SystemRandom, because
+    # os.urandom is not guaranteed to read from /dev/urandom.
+
+    def _urandom(self, size: int) -> bytes:
+        with open("/dev/urandom", "rb") as f:
+            return f.read(size)
+
+    def getrandbits(self, k: int) -> int:
+        assert k >= 0
+        size = bits_to_bytes(k)
+        n = int_from_bytes(self._urandom(size))
+        # trim excess bits
+        return n >> (size * 8 - k)
+
+    def random(self) -> float:
+        # adapted from random.SystemRandom.random
+        return (int_from_bytes(self._urandom(7)) >> 3) * (2**-53)
+
+
+class URandomProvider(HypothesisProvider):
+    # A provider which reads directly from /dev/urandom as its source of randomness.
+    # This provider exists to provide better Hypothesis integration with Antithesis
+    # (https://antithesis.com/), which interprets calls to /dev/urandom as the
+    # randomness to mutate. This effectively gives Antithesis control over
+    # the choices made by the URandomProvider.
+    #
+    # If you are not using Antithesis, you probably don't want to use this
+    # provider.
+
+    def __init__(self, conjecturedata: Optional["ConjectureData"], /):
+        super().__init__(conjecturedata)
+        self._random = URandom()
```

Comment:
I think it'd make sense to check if we're on Windows here, and... I think emit a warning but don't reassign `self._random`, so that you get the default behavior?